### ISSUE TITLE
Change and compatibility rules for config=false (state) data

### ISSUE THREAD

**Issue by BalazsLengyel** [netmod-wg/yang-next#109]
Allowed changes and what is compatible should be specified differently for config=true and config=false data.

E.g. making an optional leaf mandatory is non-backwards compatible for configuration data and thus not allowed in RFC7950. The same change for config=false should be a compatible change.

**Comment by abierman** [comment]
What did the Version design team decide about this issue?
Seems like an old-client will expect this data if madatory=true.
If changed to false then the expected data is missing from the retrieval reply

More Discuss

**Comment by janlindblad** [comment]
I agree the rules should be different for config true and config false. For config true data, it's BC to make something optional, but for config false, the opposite is true.

**Comment by reshad-equinix** [comment]
I just got PTSD (from version design team discussions) seeing this issue

**Comment by kwatsen** [comment]
Seems reasonable to clarify opposite behavior for `config false`, `output`, and `notification`
